install.py

```diff
@@ -439,6 +439,12 @@ def validate_path(path):
 
 
 def driver():
+    ninja_path = shutil.which("ninja")
+    if ninja_path is None:
```

Comment:
could use walrus here

```suggestion
    if (ninja_path := shutil.which("ninja")) is None:
```

Reply:
or alternatively, a ternary:
```
ninja_path = shutil.which("ninja")
cmake_generator_default = None if ninja_path is None else "Ninja
```